crates/anvil/core/src/eth/transaction/mod.rs

```diff
@@ -97,10 +98,10 @@ pub fn transaction_request_to_typed(tx: TransactionRequest) -> Option<TypedTrans
             }))
         }
         // EIP2930
-        (Some(1), _, None, None, _, None, None, None) |
-        (None, _, None, None, Some(_), None, None, None) => {
+        (Some(1), _, None, None, _, None, None, None, _) |
+        (None, _, None, None, Some(_), None, None, None, _) => {
```

Comment:
Can we reuse logic from alloy here somehow?

Reply:
We can probably, but we'd still have to treat `Deposit` tx case separately. In Reth we also have this logic duplicated I believe.

It's basically similar to `TransactionRequest` -> `TypedTransaction` conversion but here we're doing `unwrap_or_default` instead of throwing errors on missing fields, not sure why

Reply:
yeah this logic is very cursed -.-

we could use the TransactionRequest::build_signed here I think, if we use alloy-consensus types, we can check for OP deposit seperately

Reply:
we can do this but separately